docs/docs/cmd/spo/listitem/listitem-roleassignment-add.md

```diff
@@ -0,0 +1,81 @@
+# spo listitem roleassignment add
+
+Adds a role assignment to a listitem.
+
+## Usage
+
+```sh
+m365 spo listitem roleassignment add [options]
+```
+
+## Options
+
+`-u, --webUrl <webUrl>`
+: URL of the site where the listitem is located
+
+`--listId [listId]`
+: ID of the list. Specify either listId, listTitle or listUrl but not multiple.
+
+`--listTitle [listTitle]`
+: Title of the list. Specify either listId, listTitle or listUrl but not multiple.
+
+`--listUrl [listUrl]`
+: Relative URL of the list. Specify either listId, listTitle or listUrl but not multiple.
+
+`--listItemId <listItemId>`
+: Id of the listitem to assign the role to.
+
+`--principalId [principalId]`
+: The SharePoint Id of the principal. It may be either a user id or group id to add a role assignment for. Specify either upn, groupName or principalId.
+
+`--upn [upn]`
+: upn/email of user to assign role to. Specify either upn, groupName or principalId.
+
+`--groupName [groupName]`
+: The group name of Azure AD or SharePoint group. Specify either upn, groupName or principalId.
+
+`--roleDefinitionId [roleDefinitionId]`
+: ID of role definition. Specify either roleDefinitionId or roleDefinitionName but not both
+
+`--roleDefinitionName [roleDefinitionName]`
+: Enter the name of a role definition, like 'Contribute', 'Read', etc. Specify either roleDefinitionId or roleDefinitionName but not both
+
+--8<-- "docs/cmd/_global.md"
+
+## Examples
+
+Add role assignment to listitem _1_ in list _someList_ located in site _https://contoso.sharepoint.com/sites/project-x_for principal id _11_ and role definition id _1073741829_.
```

Comment:
we agreed not to keep values in  example descriptions. need to correct  this one